src/illuminatio/test_orchestrator.py

```diff
@@ -36,90 +46,113 @@ def getManifest(yaml_filename):
     return daemonset_manifest
 
 
-class NetworkTestOrchestrator:
-    logger = None
+def _hosts_are_in_cluster(case):
+    return all([isinstance(host, (ClusterHost, GenericClusterHost))
+                for host in [case.from_host, case.to_host]])
+
+
+# TODO refactor with above method / at least use already polled namespaces in self._currentNamespaces
+def _create_project_namespace_if_missing(api: k8s.client.CoreV1Api):
+    namespace_labels = {ROLE_LABEL: "daemon-runner-namespace", CLEANUP_LABEL: CLEANUP_ON_REQUEST}
+    namespace_list = api.list_namespace(label_selector=labels_to_string(namespace_labels))
+    if not namespace_list.items:
+        namespace = k8s.client.V1Namespace(
+            metadata=k8s.client.V1ObjectMeta(name=PROJECT_NAMESPACE, labels=namespace_labels))
+        api.create_namespace(namespace)
 
+
+class NetworkTestOrchestrator:
+    """
+    Class for handling test case related kubernetes resources
+    """
     def __init__(self, test_cases, log):
-        self._test_cases = test_cases
+        self.test_cases = test_cases
         self._current_pods = []
         self._current_services = []
-        self._current_namespaces = []
+        self.current_namespaces = []
         self.runner_daemon_set = None
-        self.runner_image = ""
-        self.target_image = ""
-        global logger
-        logger = log
+        self.docker_image = {}
+        self.logger = log
 
     def set_runner_image(self, runner_image):
-        self.runner_image = runner_image
+        """
+        Updates the runner docker image
+        """
+        self.docker_image["runner"] = runner_image
```

Comment:
Please use `runner_image` or `oci_image` in the end we don't have to use a `Docker image` at all.